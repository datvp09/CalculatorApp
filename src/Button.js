import React from "react"
import { View, TouchableHighlight, Text } from "react-native"
import styles from "./styles"

const Button = React.memo(props => (
  <TouchableHighlight
    style={[styles.buttonStyles, props.style]}
    onPress={props.onPress}
    underlayColor="#777"
  >
    <View style={[styles.contButton, { borderRadius: 5 }]}>
      <Text style={styles.txtButtons}>{props.text}</Text>
    </View>
  </TouchableHighlight>
))

export default Button
