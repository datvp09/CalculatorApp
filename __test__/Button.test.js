import "react-native"
import React from "react"
import Button from "../src/Button"
import renderer from "react-test-renderer"

describe("Button", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const component = renderer
        .create(<Button text={"9"} onPress={() => {}} style={{}} />)
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
