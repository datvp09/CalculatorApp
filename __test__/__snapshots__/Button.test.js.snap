// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "flex": 1,
        "marginBottom": 10,
        "marginLeft": 10,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ddd",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "borderRadius": 5,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 26,
        }
      }
    >
      9
    </Text>
  </View>
</View>
`;
